### PR TITLE
Dynamic Fees

### PR DESCRIPTION
**Pull request by voloshyn**
### Motivation

Implement [dynamic fees](https://github.com/mobilecoinfoundation/rfcs/blob/main/text/0001-dynamic-fees-v1.md)

### In this PR
* Add `BlockchainClient` to fetch and cache the current minimum transaction fee
* Remove fee levels
* New method `getOrFetchMinimumTxFee` added to `MobileCoinClient`

### CODE REVIEW

**Review by sugargoat — APPROVED** [review]

File: android-sdk/src/main/java/com/mobilecoin/lib/BlockchainClient.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import com.mobilecoin.lib.uri.ConsensusUri;
+import com.mobilecoin.lib.util.NetworkingCall;
+
+import java.math.BigInteger;
+
+import consensus_common.BlockchainAPIGrpc;
+import consensus_common.ConsensusCommon;
+import io.grpc.StatusRuntimeException;
+
+class BlockchainClient extends AnyClient {
```
Comment (samdealy) [inline android-sdk/src/main/java/com/mobilecoin/lib/BlockchainClient.java:18]: Can this class be made final?

Comment (samdealy) [inline android-sdk/src/main/java/com/mobilecoin/lib/BlockchainClient.java:18]: Can you add a JavaDoc for this class that describes what it does?

File: android-sdk/src/main/java/com/mobilecoin/lib/BlockchainClient.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+     */
+    BlockchainClient(@NonNull ConsensusUri uri,
+                     @NonNull ClientConfig.Service serviceConfig) {
+        super(uri.getUri(), serviceConfig);
+    }
+
+    /**
+     * Fetch or return cached current minimal fee
+     */
+    synchronized
```
Comment (samdealy) [inline android-sdk/src/main/java/com/mobilecoin/lib/BlockchainClient.java:37]: nit: make same line as line 38

File: android-sdk/src/main/java/com/mobilecoin/lib/BlockchainClient.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+     * Reset cache
+     */
+    synchronized void resetCache() {
+        lastBlockInfo = null;
+    }
+
+    /**
+     * Fetch or return cached last block info
+     */
+    synchronized
```
Comment (samdealy) [inline android-sdk/src/main/java/com/mobilecoin/lib/BlockchainClient.java:58]: Same as above.

**Review by samdealy — APPROVED** [review]

**Pull request merged**